src/sync_session.cpp

```diff
@@ -230,7 +255,9 @@ struct sync_session_states::Dying : public SyncSession::State {
 
     void log_out(std::unique_lock<std::mutex>& lock, SyncSession& session) const override
     {
-        session.advance_state(lock, inactive);
+        if (session.m_pending_download_threads == 0 && session.m_pending_upload_threads == 0) {
+            session.advance_state(lock, inactive);
+        }
```

Comment:
This is `dying`'s log out function, there's an assertion that forbids advancing a state to itself. (We also don't want to rerun the 'enter state' work if the binding asks the session to log out while it's already in the process of dying.)